sklearn/conftest.py

```diff
@@ -233,25 +233,22 @@ def pyplot():
     pyplot.close("all")
 
 
-def pytest_runtest_setup(item):
```

Comment:
Maybe it would be a bit simpler to `pytest_sessionstart` (or another more relevant hooks) rather than a fixture with autouse?

See https://docs.pytest.org/en/7.2.x/reference/reference.html#pytest.hookspec.pytest_sessionstart

Reply:
https://github.com/scikit-learn/scikit-learn/pull/25918#issue-1632468306

I tried, but it's not executed in the CI. I could reproduce locally, it has to do with running from a separate test folder and with ``--pyargs``

Reply:
Maybe https://github.com/pytest-dev/pytest/issues/1596, that seems to indicate that when using `--pyargs` `conftest.py` is not discovered somehow?

Reply:
`pytest_configure` seems to work on the other hand from some small tests, so maybe use it instead?

Reply:
A "session-scoped, auto-used fixture" is not a that easy to grasp concept compared to procedural code put in a function named `pytest_configure`. Personally I am fine with either option if they both yield the same result with a slight +0.5 for `pytest_configure`.